MobiusCore/Source/Mobius.swift

```diff
@@ -19,7 +19,26 @@
 
 import Foundation
 
-public typealias Update<Model, Event, Effect> = (Model, Event) -> Next<Model, Effect>
+/// A wrapper around an update function.
+public struct Update<Model, Event, Effect> {
+    private let update: (inout Model, Event) -> [Effect]
+
+    public init(_ update: @escaping (inout Model, Event) -> [Effect]) {
+        self.update = update
+    }
+
+    public static func create(_ update: @escaping (Model, Event) -> Next<Model, Effect>) -> Update {
```

Comment:
This is to have some form of backwards compatibility. Existing update functions can wrap a signature of the old style by just doing `Update.create { model, event in ... }`. We cannot simply overload the initializer of `Update` to use both styles of signature because it will lead to some ambiguous type checking errors.